lib/content/message/predictions.ex

```diff
@@ -154,17 +151,100 @@ defmodule Content.Message.Predictions do
     end
   end
 
-  defp calculate_crowding_data_confidence(_prediction, nil), do: nil
+  defp do_crowding(prediction) when prediction.route_id in ["Orange"] do
+    case Engine.Locations.for_vehicle(prediction.vehicle_id) do
+      %Locations.Location{} = location ->
+        {calculate_crowding_data_confidence(prediction, location),
+         get_crowding_description(location.multi_carriage_details)}
+
+      _ ->
+        {nil, nil}
+    end
+  end
+
+  defp do_crowding(_), do: {nil, nil}
 
   defp calculate_crowding_data_confidence(prediction, location)
-       when prediction.route_id in ["Orange"] and location.stop_id == prediction.stop_id do
+       when location.stop_id == prediction.stop_id do
     if location.status in [:incoming_at, :in_transit_to],
       do: :high,
       else: :low
   end
 
   defp calculate_crowding_data_confidence(_prediction, _location), do: nil
 
+  defp get_crowding_description([]), do: nil
+
+  defp get_crowding_description(carriage_details) do
+    crowding_levels =
+      Enum.map(carriage_details, &occupancy_status_to_crowding_level(&1.occupancy_status))
+
+    min_crowding_level = Enum.min(crowding_levels)
+
+    relative_crowding_levels =
+      for crowding_level <- crowding_levels do
+        if crowding_level == min_crowding_level,
+          do: :e,
+          else: :f
+      end
+
+    {get_emptier_location(
+       {Enum.count(relative_crowding_levels, &Kernel.==(&1, :e)), relative_crowding_levels}
+     ), crowding_level_to_atom(min_crowding_level)}
+  end
+
+  defp occupancy_status_to_crowding_level(occupancy_status) do
+    case occupancy_status do
+      :many_seats_available -> 1
+      :few_seats_available -> 1
+      :standing_room_only -> 2
+      :crushed_standing_room_only -> 3
+      :full -> 3
+      _ -> 4
+    end
+  end
+
+  defp crowding_level_to_atom(crowding_level) do
+    case crowding_level do
+      1 -> :not_crowded
+      2 -> :some_crowding
+      3 -> :crowded
+      4 -> :unknown_croding
+    end
+  end
+
+  defp get_emptier_location(car_crowding_levels) do
+    case car_crowding_levels do
+      {1, [_, _, :f, :f, :f, :f]} -> :front
+      {1, [:f, :f, :f, :f, _, _]} -> :back
+      {1, [:f, :f, _, _, :f, :f]} -> :middle
+      {2, [_, _, _, :f, :f, :f]} -> :front
+      {2, [_, :f, :f, _, :f, :f]} -> :front
+      {2, [:f, :f, :f, _, _, _]} -> :back
+      {2, [:f, :f, _, :f, :f, _]} -> :back
+      {2, [:f, :f, _, _, :f, :f]} -> :middle
+      {2, [:f, _, :f, _, :f, :f]} -> :middle
+      {2, [:f, :f, _, :f, _, :f]} -> :middle
```

Comment:
Looks like `2` needs case(s) for `:front_and_back`. Check my logic, but I think we could collapse things a bit by relying on the fact that these are checked in order, so `[_, _, :f, :f, _, _]` would suffice as long as it came after `:front` and `:back`. Similarly, the `:middle` cases could be handled with just `[:f, _, _, _, _, :f]` as the last check (or just use a default).

Reply:
Ah good catch. I meant to add a default to catch the `:front_and_back` case but must've skipped it. I think you're right though about collapsing. Will check the logic on that and update the PR and merge if all looks good